packages/plugin-core/src/workspace/workspaceActivator.ts

```diff
@@ -581,8 +584,16 @@ export class WorkspaceActivator {
       return new EngineNoteProvider(engine);
     };
     if (!opts?.skipTreeView) {
```

Comment:
why can't we do this in `onWorkspaceOpen`? if we need for state to be initialized, would be better to introduce a new hook (eg. `onWorkspaceActivate` then add initializer specific logic into `workspaceActiavtor`

Reply:
Are you suggesting we move the tree view init step itself to each respective workspace initializers?

Reply:
I left the tree view init logic in the activator since the only specific logic here would be the special treatment with the tutorial workspace after it was initialized.